index.js

```diff
@@ -40,14 +42,24 @@ var assertionVisitor = {
 				return;
 			}
 
+			var hoisted = hoistAwaitAndYield(path, arg0);
+
 			path.node.arguments[0] = wrapWithHelper({
 				HELPER_ID: t.identifier(this.avaThrowHelper()),
-				EXP: arg0,
+				EXP: hoisted ? hoisted.arg0 : arg0,
 				LINE: t.numericLiteral(arg0.loc.start.line),
 				COLUMN: t.numericLiteral(arg0.loc.start.column),
 				SOURCE: t.stringLiteral(state.file.code.substring(arg0.start, arg0.end)),
 				FILE: t.stringLiteral(state.file.opts.filename)
 			}).expression;
+
+			if (hoisted) {
+				path.replaceWithMultiple(
+					hoisted.expressions.concat(
+						t.expressionStatement(path.node)
```

Comment:
You are replacing the Callxpression by an ExpressionStatement here. Would the following test code stll work, or would Babel crash or something because you are building an odd AST?
```js
test('should foo...', async t => {
  t.is(
      t.throws(foo(await quux())).message,
     'Could not find X'
   );
});
```
My worry is that you can't AFAIK pass a statement as the argument of a function (here, the `t.is(){ call). But I don't know how Babel would react to that. Would you mind adding a test case and see how that pans out?
I would be fine with not entirely supporting this edge case, but it should at the very least not crash.